core/commands/storage/challenge.go

```diff
@@ -138,18 +143,25 @@ func (sc *StorageChallenge) GenChallenge() error {
 	return nil
 }
 
-// SolveChallenge solves the given challenge block hash + nonce's request
+// SolveChallenge solves the given challenge chunk index + nonce's request
 // and records the solved/responding value in Hash field.
-func (sc *StorageChallenge) SolveChallenge(chHash cid.Cid, chNonce string) error {
+func (sc *StorageChallenge) SolveChallenge(chIndex int, chNonce string) error {
```

Comment:
has a little concern for https://github.com/TRON-US/go-btfs/pull/251/files#diff-2db60460fd212471936d893e5c5da829L77   not sure why chunk can be re-seen , is that mean sometime the chunk can be parsed on different sequence?

Reply:
This is to generate the StorageChallenge once because it involves some DAG tree parsing that can be slow, then on each challenge from guard it can reuse the object to compute the answer fast.